docs: update tanstack query integration documentation to the new api

kalijonn opened the pull request:
## Related Issues or Discussions

Related: https://github.com/jotaijs/jotai-tanstack-query/pull/45

## Summary



## Check List

- [x] `yarn run prettier` for formatting code and docs

kalijonn commented:
I'll update csbs after releasing v.0.8.0

dai-shi reviewed (commented):
I think we need migration guide to v0.8.0.
Either a section in this doc, or a link to a doc in jotai-tanstack-query repo is fine.

kalijonn commented:
> I think we need migration guide to v0.8.0. Either a section in this doc, or a link to a doc in jotai-tanstack-query repo is fine.

I added a migration guide. Can you review the level of detail? Too detailed or not detailed enough?

I also released jotai-tanstack-query v0.8.0 and added updated the sandboxes here.

dai-shi commented on docs/extensions/query.mdx:
```diff
@@ -7,53 +7,69 @@ keywords: tanstack,query
 
 const idAtom = atom(1)
-const [userAtom] = atomsWithQuery((get) => ({
+const userAtom = atomsWithQuery((get) => ({
```
```suggestion
const userAtom = atomWithQuery((get) => ({
```

dai-shi commented on docs/extensions/query.mdx:
````diff
@@ -62,91 +78,169 @@ const [userAtom] = atomsWithQuery((get) => ({
 
 > Unlike queries, mutations are typically used to create/update/delete data or perform server side-effects.
 
 ```jsx
````
```suggestion
```tsx
```

dai-shi reviewed (commented):
I think the migration guide is good.
